### DISCUSSION TITLE
Sailor Hat WiFi Gateway

### DISCUSSION THREAD

**Discussion by mairas** [hatlabs/discussions#35] [Show and tell]
Dear all,

I have now finished and released for sale the Sailor Hat WiFi gateway (SH-wg). SH-wg transmits NMEA 2000 traffic over WiFi. It can be used to transmit N2K data to popular navigation and instrumentation apps such as Navionics Boating, iSailor and WilhelmSK. Other use cases include setting up wireless bridges between two separate NMEA 2000 segments, and implementing wireless NMEA 2000 interfaces for compatible applications such as the Signal K server.

SH-wg is open hardware and open-source, just like all other Hat Labs products.

SH-wg is immediately available at the Hat Labs website.

https://hatlabs.fi/product/sh-wg/
https://docs.hatlabs.fi/sh-wg/

**Comment by ba58smith** [comment]
Hey, Matti! It's been a long time - hope you are well. My wife and I are joining friends on their sailboat from Panama to French Polynesia (with a stop in the Galapagos)  - should be fun! I'm looking at https://docs.hatlabs.fi/sh-wg/pages/getting_started/ to see if I might be able to use it on his boat. His boat doesn't have a wifi network, but if I configure a SH-WG as an Access Point, I should be able to get data from it on Navionics Boating, right? And that might be enough. 

But I'd like to get the data into SK. However, the Getting Started document makes me think that won't work without an already-existing wifi network, for both the SH-WG and the SK Server (it would be a Pi) to connect to. But maybe, when writing the documentation, you just assumed a SK Server would be connecting to a wifi network already? So the question is, if I set up the SH-WG as an Access Point, can I have the SK Server Pi connect to it, and then SK will be available on the boat via the SH-WG's wifi network? Thanks!

**Reply by mairas** [reply]
Hi Butch! Thanks for asking - well and busy! Sounds like a fantastic trip ahead!

You're right that I didn't think of the prospect of using the SH-wg as an Access Point for an SK server. 😄 But now that you mention it, it should definitely work. However, if you intend to keep the SK server running continuously, I'd rather make that provide the WiFi AP and make the other devices connect to it. For example OpenPlotter can do that out of the box, and that approach makes for a more useful WiFi. The AP created by the SH-wg is quite slow - totally fine for the data transmissions the device itself is performing, but for example accessing the Pi through it could be annoying.

**Reply by ba58smith** [reply]
Ah, of course. There will be no Internet on the boat (at least not while we're crossing the Pacific), but that doesn't mean I can't make the SK Pi into a wifi "server". Thanks - the SH-WG looks like a really cool gadget!
